kubens

```diff
@@ -58,7 +62,19 @@ current_context() {
 }
 
 get_namespaces() {
-  $KUBECTL get namespaces -o=jsonpath='{range .items[*].metadata.name}{@}{"\n"}{end}'
+  name_jsonpath='{range .items[*].metadata.name}{@}{"\n"}{end}'
+  namespaces=$($KUBECTL get namespaces -o=jsonpath="${name_jsonpath}" 2>/dev/null)
+  if [[ $? -eq 0 ]]; then
+    echo "${namespaces}"
+    return 0
+  fi
+
+  $KUBECTL auth can-i get namespaces --quiet 1>&$STDOUT 2>&$STDERR <&$STDIN
+  if [[ $? -ne 0 ]]; then
+    return $?
+  fi
+
+  $KUBECTL get namespaces -o=jsonpath="${name_jsonpath}"
```

Comment:
what's the point of repeating this? (making the `stderr` visible to the user?)

Reply:
If the `kubectl auth can-i get namespaces` successfully triggered an auth plugin above then this `kubectl get namespaces` call should now succeed and return the namespace output that `get_namespaces` is supposed to provide